test/crdsvalidation/konnectgatewaycontrolplane_test.go

```diff
@@ -233,6 +233,27 @@ func TestKonnectGatewayControlPlane(t *testing.T) {
 					kcp.Spec.KonnectConfiguration.APIAuthConfigurationRef.Name = "name-2"
 				},
 			},
+			{
+				Name: "cluster_type change is not allowed",
```

Comment:
Can we also cover the rest of the added rules (members can only be set when `CLUSTER_TYPE_CONTROL_PLANE_GROUP`, `spec.konnect.authRef` is immutable)?

Reply:
Tests for setting members are already covered in https://github.com/Kong/kubernetes-configuration/blob/b2418da9f1f569aaf882f9b5d7682ed132295eb9/test/crdsvalidation/konnectgatewaycontrolplane_test.go#L19-L132

Tests for `konnect.aurhRef` change can be found in https://github.com/Kong/kubernetes-configuration/blob/b2418da9f1f569aaf882f9b5d7682ed132295eb9/test/crdsvalidation/konnectgatewaycontrolplane_test.go#L138-L235.

Do you have something else in mind?

Reply:
🤦 Sorry, I misread the diff and thought these I mentioned were added.